src/main/webapp/app/exercises/programming/manage/update/update-components/information/programming-exercise-information.component.ts

```diff
@@ -7,43 +7,90 @@ import { Subject, Subscription } from 'rxjs';
 import { TableEditableFieldComponent } from 'app/shared/table/table-editable-field.component';
 import { every } from 'lodash-es';
 import { ImportOptions } from 'app/types/programming-exercises';
+import { ProgrammingExerciseInputField } from 'app/exercises/programming/manage/update/programming-exercise-update.helper';
+import { removeSpecialCharacters } from 'app/shared/util/utils';
 
 @Component({
     selector: 'jhi-programming-exercise-info',
     templateUrl: './programming-exercise-information.component.html',
-    styleUrls: ['../../programming-exercise-form.scss', 'programming-exercise-information.component.scss'],
+    styleUrls: ['../../../programming-exercise-form.scss', 'programming-exercise-information.component.scss'],
 })
-export class ProgrammingExerciseInformationComponent implements AfterViewInit, OnDestroy {
+export class ProgrammingExerciseInformationComponent implements AfterViewInit, OnChanges, OnDestroy {
+    protected readonly ProjectType = ProjectType;
+
     @Input() isImport: boolean;
     @Input() isExamMode: boolean;
     @Input() programmingExercise: ProgrammingExercise;
     @Input() programmingExerciseCreationConfig: ProgrammingExerciseCreationConfig;
     @Input() isLocal: boolean;
     @Input() importOptions: ImportOptions;
+    isSimpleMode = input.required<boolean>();
+    isEditFieldDisplayedRecord = input.required<Record<ProgrammingExerciseInputField, boolean>>();
 
     @ViewChild(ExerciseTitleChannelNameComponent) exerciseTitleChannelComponent: ExerciseTitleChannelNameComponent;
     @ViewChildren(TableEditableFieldComponent) tableEditableFields?: QueryList<TableEditableFieldComponent>;
     @ViewChild('shortName') shortNameField: NgModel;
     @ViewChild('checkoutSolutionRepository') checkoutSolutionRepositoryField?: NgModel;
     @ViewChild('recreateBuildPlans') recreateBuildPlansField?: NgModel;
     @ViewChild('updateTemplateFiles') updateTemplateFilesField?: NgModel;
+    @ViewChild('titleChannelNameComponent') titleComponent?: ExerciseTitleChannelNameComponent;
```

Comment:
you could also use signals for the view child here :) `viewChild()`

Reply:
I adjusteed it where it was easily possible without further changes :)